src/utils/error.hpp

```diff
@@ -31,15 +31,17 @@ namespace detail {
     // Lightweight std::source_location.
     struct source_location {
         const char* const file;
-        //const char* const function; // disabled for now due to static constexpr restrictions
+        // const char* const function; // disabled, because there are many different compiler definitions for it (see for example BOOST_CURRENT_FUNCTION)
```

Comment:
I think this may as well just be removed, it's not doing anyone any good being commented out. Could you do me a favor and remove this while you're here?